src/Stl/Caching/FileSystemCache.cs

```diff
@@ -13,10 +13,21 @@ namespace Stl.Caching
     public abstract class FileSystemCacheBase<TKey, TValue> : AsyncCacheBase<TKey, TValue>
         where TKey : notnull
     {
+#if !NETSTANDARD2_0
+        private const int BufferSize = -1;
+#else
+        private const int MinAllowedBufferSize = 128;
+        private const int BufferSize = MinAllowedBufferSize;
+#endif
+        
         public override async ValueTask<Option<TValue>> TryGet(TKey key, CancellationToken cancellationToken = default)
         {
             try {
+                #if !NETSTANDARD2_0
                 await using var fileStream = OpenFile(GetFileName(key), false, cancellationToken);
+                #else
+                using var fileStream = OpenFile(GetFileName(key), false, cancellationToken);
```

Comment:
Try adding .AsAsyncDisposable() to XxxStream, XxxStreamWriter, ...

Reply:
Added AsAsyncDisposable. With it code takes now 2 lines instead of 5 lines with conditional compilation.
It's used in FileSystemCache file and some other places together with cancellationToken.Register() method.
Please check if it's ok.